src/EventArgs.cs

```diff
@@ -0,0 +1,436 @@
+﻿namespace Microsoft.IO
+{
+    using System;
+
+    public sealed partial class RecyclableMemoryStreamManager
+    {
+        /// <summary>
+        /// Arguments for the StreamCreated event
+        /// </summary>
+        public struct StreamCreatedEventArgs
```

Comment:
Would making these structs readonly be a limiting factor for future API updates? Otherwise, I think these are all candidates to be readonly.